GVFS/GVFS.Notifications/VFSForGit.Mac/StatusMenuItem/Utilities/VFSProcessRunner.m

```diff
@@ -0,0 +1,85 @@
+#import "VFSProcessRunner.h"
+
+@interface VFSProcessRunner()
+
+@property (strong) ProcessFactory processFactory;
+
+@end
+
+@implementation VFSProcessRunner
+
+- (instancetype _Nullable)initWithProcessFactory:(ProcessFactory _Nonnull)processFactory
+{
+    if (self = [super init])
+    {
+        if (!processFactory)
+        {
+            self = nil;
+        }
+        else
+        {
+            _processFactory = processFactory;
+        }
+    }
+    
+    return self;
+}
+
+/**
+ Runs an executable specified by path and args. If the executable could be run
+ successfully - output will contain the executable's combined stderr/stdout
+ output and the method returns YES. In case of failure, it returns NO and error
+ will hold the executable's combined stderr/stdout output.
+
+ @param path - specify full path to the executable.
+ @param args - specify any command line args to pass to the executable.
+ @param output - contains executable's output, if it was successfully run.
+ @param error - contains executable's output, if it exited with an error.
+ @return YES if the executable was successfully run, NO otherwise.
+ */
+- (BOOL)tryRunExecutable:(NSURL *_Nonnull)path
+                    args:(NSArray<NSString *> *_Nullable)args
+                  output:(NSString *_Nonnull *_Nonnull)output
+                   error:(NSError *_Nonnull *_Nonnull)error
```

Comment:
Should be (NSError *_Nonnull *_Nullable) if it is a required parameter=nonnull, and the second part is nullable because the output value itself (=NSError*) can be nil in the success scenario.

Reply:
Updated. For NSError using the default nullability. For other out parameters using `(*_Nullable *_Nonnull)`.

Example: 
`- (BOOL)tryRunExecutable:(NSURL *)path
                    args:(NSArray<NSString *> *_Nullable)args
                  output:(NSString *_Nullable __autoreleasing *_Nonnull)output
                   error:(NSError * __autoreleasing *)error;
`